engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/HashCodeNode.java

```diff
@@ -486,12 +487,7 @@ long hashCodeForString(
     } catch (UnsupportedMessageException e) {
       throw new IllegalStateException(e);
     }
-    Normalizer2 normalizer = Normalizer2.getNFDInstance();
-    if (normalizer.isNormalized(str)) {
-      return str.hashCode();
-    } else {
-      return normalizer.normalize(str).hashCode();
-    }
+    return Core_Text_Utils.unicodeNormalizedHashCode(str);
```

Comment:
`hashCodeForString` is behind `@TruffleBoundary` already. The change seems safe.